evpp/httpc/conn_pool.cc

```diff
@@ -42,6 +52,8 @@ void ConnPool::Put(const ConnPtr& c) {
         return;
     }
     it->second.push_back(c);
+    // Take over the ownership
+    c.reset();
```

Comment:
We don't think this is a good refactoring. `Put(...)` method means place something in a pool, but do not change it.

Reply:
connection seems to be resource borrowed from connection pool,  so when put back it needs to take over the resource.  I can remove the reset,  but that would mean every place we Put,  we need to reset.  For example:

if (pool_) {
    pool_->Put(conn_);
}

will need to become
if (pool_) {
    pool_->Put(conn_);
    conn_.reset();
}

Which can be error prone as everyone who does Put needs to remember to reset.

I can make that change to const and remove reset but wanted to make sure we are doing the right tradeoff here.  Let me know.

Reply:
I still think the original code `void ConnPool::Put(const ConnPtr& c) ` is better. Please take a look. Thanks.